### PR TITLE
[GenerationOutputs] Fix GenerationOutputs Tests 

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

The `GenerationOutputs` PR: https://github.com/huggingface/transformers/pull/9150 was not rebased, so that the cicrle ci on master is red now. This PR fixes it.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors which may be interested in your PR.

### CODE REVIEW

File: src/transformers/generation_utils.py [hunk 1]
```diff
@@ -191,6 +191,7 @@ class BeamSearchDecoderOnlyOutput(ModelOutput):
             Processed beam scores for each vocabulary token at each generation step. Beam scores consisting of log
             softmax scores for each vocabulary token and sum of log softmax of previously generated tokens in this beam
             . :obj:`(max_length,)`-shaped tuple of :obj:`torch.FloatTensor` with each tensor of shape :obj:`(batch_size
+
```
Comment (sgugger) [inline src/transformers/generation_utils.py:194]: Ugh, another failure of the new doc styler. Can we use no space for the multiplications so that it doesn't believe this a listt?

Comment (patrickvonplaten) [reply src/transformers/generation_utils.py:194]: yes will change it :-)

File: tests/test_generation_utils.py [hunk 2]
```diff
@@ -522,6 +522,7 @@ def test_greedy_generate_dict_outputs_use_cache(self):
                 return
 
             config.use_cache = True
+            config.is_decoder = True
```
Comment (patrickvonplaten) [inline tests/test_generation_utils.py:525]: make sure to use causal mask

File: src/transformers/models/encoder_decoder/modeling_encoder_decoder.py [hunk 3]
```diff
@@ -455,7 +455,7 @@ def prepare_inputs_for_generation(
             "decoder_attention_mask": decoder_attention_mask,
             "decoder_input_ids": decoder_inputs["input_ids"],
             "encoder_outputs": encoder_outputs,
-            "past_key_values": past,
+            "past_key_values": decoder_inputs["past_key_values"],
```
Comment (patrickvonplaten) [inline src/transformers/models/encoder_decoder/modeling_encoder_decoder.py:458]: @patil-suraj -> think this is a bit safer

File: src/transformers/models/bert_generation/modeling_bert_generation.py [hunk 4]
```diff
@@ -570,7 +570,7 @@ def prepare_inputs_for_generation(self, input_ids, past=None, attention_mask=Non
         if past is not None:
             input_ids = input_ids[:, -1:]
 
-        return {"input_ids": input_ids, "attention_mask": attention_mask}
+        return {"input_ids": input_ids, "attention_mask": attention_mask, "past_key_values": past}
```
Comment (patrickvonplaten) [inline src/transformers/models/bert_generation/modeling_bert_generation.py:573]: @patil-suraj we forgot to add this in the BERT cache PR. Bert-like models can also be used as stand-alone `BertForCausalLM` models -> so we need to return `past_key_values` here.

**Comment by patrickvonplaten** [comment]
This PR actually made me correct 2 bugs additionally:

1) past_key_values for BertForCausalLM
2) T5 don't return T5 cross attentions if just decoder model

**Review by sgugger — APPROVED** [review]
Thanks for fixing!
